geographically-distributed-postgresql.mdx

```diff
@@ -0,0 +1,178 @@
+
+In the above, the tables are presented with their schema name. In the data nodes, we assume the schema name is `puplic`. When mounting the tables to the controller node, we use a schema name that matches the region name, or the node name. This is a convenient way to keep track of where each table is coming from.
+
+For example, here are the commands to mount the `pages` table from the `us_east1` node to the controller node:
+
+First, define the remote server:
+
+```sql
+CREATE EXTENSION postgres_fdw;
+
+CREATE SERVER us_east1
+  FOREIGN DATA WRAPPER postgres_fdw
+    OPTIONS (host 'us_east1', dbname 'notionclone');
+
+CREATE USER MAPPING FOR appuser
+  SERVER us_east1
+    OPTIONS (user 'appuser', password 'mypassword');
+
+```
+
+Import remote tables, all at once, in a dedicated schema:
+
+```sql
+CREATE SCHEMA us_east1;
+IMPORT FOREIGN SCHEMA public FROM SERVER us_east1 INTO us_east1;
+```
+
+You can now query use the `us_east1.pages` table as if it was a local table.
+
+After you have done this for all the other nodes/regions, you can also create a partitioned table that will act as the global view:
+
+```sql
+CREATE SCHEMA global;
+CREATE TABLE global.pages(
+  id TEXT,
+  title TEXT,
+  url TEXT,
+  wsId TEXT,
+  region TEXT) PARTITION BY LIST(region);
+
+ALTER TABLE global.pages
+  ATTACH PARTITION us_east1.pages
+    FOR VALUES IN ('us-east-1');
+
+ALTER TABLE global.pages
+  ATTACH PARTITION eu_central1.pages
+    FOR VALUES IN ('eu-central-1');
+
+ALTER TABLE global.pages
+  ATTACH PARTITION ap_southeast2.pages
+    FOR VALUES IN ('ap-southeast-2');
+```
+
+You can query the `global.pages` table as if it was a global view of the full data.
+
+You can also insert data via the `global.pages` table, as long as you provide a value for the `region` column (or have a default set on the column). The data will be routed to the correct node.
+
+While for simplicity we have used a separate controller node, you can also create the global views in any or all of the data nodes. In this case, you can use both local and remote tables as partitions of the same partitioned table.
+
+### Limitations
+
+While the FDW + partitioning trick (for lack of a better word) provides a global view that you can query and write to, it does have some important limitations at the moment. Note that these limitations apply only when you are accessing or writing via the global view. That's why we're saying that this pattern works best, for now, when you rarely need to use the global view.
+
+**Performance impact: it's complicated**. Accessing a remote table is a lot slower than accessing a local table, especially when the table is half-way across the globe. Also, transferring a lot of data over the Internet is slow and expensive, so that becomes another optimization metric. In order to create the optimal plans, the Postgres planner needs to be aware of these things and take it into account. As of today (Postgres 16) however, this unfortunately only partially the case.
```

Comment:
`As of today (Postgres 16) however, this is unfortunately only partially the case`?

Reply:
I think some more details here would be useful or linked references that go deeper